src/onPreprocess/defineMeasureData.js

```diff
@@ -0,0 +1,27 @@
+import { nest } from 'd3';
+
+export default function defineMeasureData() {
+    this.measure_data = this.raw_data.filter(
+        d => d[this.config.measure_col] === this.currentMeasure
+    );
+    this.filtered_measure_data = this.measure_data.filter(d => {
+        let filtered = false;
+
```

Comment:
Hate having to re-write the filter functionality here - I think it's slightly cleaner to filter `raw_data` based on the measure, and save the actual raw data in `super_raw` in the init callback. That way the regular webcharts filtering workflow still works.  Maybe not worth fixing in v2.2 (since I'm sure this works fine), but something to keep in mind for later. 

If we ever do decide to tackle webcharts v2.0 I definitely think that we should add a `preprocessed_data` array (or something like that) to the chart object. That new array would sits between the loaded data and the data that is sent to `webcharts.draw()`.

Reply:
The issue here is that we want only the measure filter to change the domains.  It'd be nice to specify in the filter settings the x- and y-behavior.  What do you think?

It'd be nice if we could access the data between each of these steps:

1. raw data
2. valid data, removing x- and y-values that don't make sense
3. filtered data, applying the filters
4. mark data, nesting the data